File: ColdDream.Api/Models/Inspiration.cs
using System.ComponentModel.DataAnnotations;

namespace ColdDream.Api.Models;

public class Inspiration
{
    public Guid Id { get; set; }
    
    [Required]
    public string ImageUrl { get; set; } = string.Empty;
    
    [Required]
    public string Description { get; set; } = string.Empty;
    
    public Guid UserId { get; set; }
    public string UserName { get; set; } = string.Empty;
    public string UserAvatar { get; set; } = string.Empty;
    
    public int Likes { get; set; }
    public int Collects { get; set; }
    
    public bool IsLiked { get; set; } // Not mapped to DB, used for DTO
    public bool IsCollected { get; set; } // Not mapped to DB, used for DTO
    
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

File: ColdDream.Web/src/api/inspiration.ts
import api, { type ApiResponse } from './index';

export interface Inspiration {
  id: string;
  imageUrl: string;
  description: string;
  userId: string;
  userName: string;
  userAvatar: string;
  likes: number;
  collects: number;
  isLiked: boolean;
  isCollected: boolean;
}

export const getInspirations = () => {
  return api.get<any, ApiResponse<Inspiration[]>>('/inspiration');
};

export const getCollectedInspirations = () => {
  return api.get<any, ApiResponse<Inspiration[]>>('/inspiration/collected');
};

export const likeInspiration = (id: string) => {
  return api.post<any, ApiResponse<any>>(`/inspiration/${id}/like`);
};

export const collectInspiration = (id: string) => {
  return api.post<any, ApiResponse<any>>(`/inspiration/${id}/collect`);
};

export const createInspiration = (data: { description: string; imageUrl: string }) => {
  return api.post<any, ApiResponse<Inspiration>>('/inspiration', data);
};
